crates/supervisor/core/src/reorg/task.rs

```diff
@@ -4,29 +4,41 @@ use alloy_primitives::{B256, ChainId};
 use alloy_rpc_client::RpcClient;
 use alloy_rpc_types_eth::Block;
 use derive_more::Constructor;
+use kona_interop::DerivedRefPair;
 use kona_supervisor_storage::{DbReader, StorageRewinder};
 use std::sync::Arc;
 use tracing::{debug, info, trace, warn};
 
 /// Handles reorg for a single chain
 #[derive(Debug, Constructor)]
-pub(crate) struct ReorgTask<DB> {
+pub(crate) struct ReorgTask<C, DB> {
```

Comment:
this change doesn't match this metrics scope, could you pls commit in a separate pr